src/rest.rs

```diff
@@ -36,29 +37,71 @@ impl Client {
         Ok(Client {
             config,
             address,
-            token: token,
+            token,
             rest: client,
         })
     }
 
-    pub async fn req<T: Serialize + ?Sized>(
+    pub async fn req<T: Serialize + ?Sized, Q: Serialize + ?Sized>(
         &self,
         url: String,
         method: Method,
         body: Option<&T>,
+        query: Option<&Q>,
```

Comment:
This feels odd. Consider switching all of these options to a `RequestBuilder`-like type. That way you can do:

```rust
client.request(url, method).body(body).query(query).headers(headermap).execute()
```

This would allow you to drop the options in the external API and handle them purely internally